pkg/api/types.go

```diff
@@ -219,6 +219,8 @@ type VolumeSource struct {
 	PersistentVolumeClaim *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
 	// RBD represents a Rados Block Device mount on the host that shares a pod's lifetime
 	RBD *RBDVolumeSource `json:"rbd,omitempty"`
+	// Metadata represents metadata about the pod that should populate this volume
+	Metadata *MetadataVolumeSource `json:"metadata"`
```

Comment:
I'm commenting on api/types.go, but the same comments also apply to v1/types.go.

Must be tagged omitempty, since it's optional.